runtime/compilers/rillv1/parser.go

```diff
@@ -259,14 +273,52 @@ func (p *Parser) Reparse(ctx context.Context, paths []string) (*Diff, error) {
 		}
 	}
 
+	// Capture if rill.yaml will be updated
+	modifiedRillYAML := p.RillYAML == nil
+
 	// Phase 2: Parse (or reparse) the related paths, adding back resources
 	err := p.parsePaths(ctx, parsePaths)
 	if err != nil {
 		return nil, err
 	}
 
+	// Infer or re-infer refs for...
+	inferRefsSeen := make(map[ResourceName]bool)
```

Comment:
nit: perhaps move inference this to a separate method?

Reply:
Thanks, it's only used in Reparse and also depends on a variable in its scope, so keeping it like this for now